### DISCUSSION TITLE
Resend Verification Email

### DISCUSSION THREAD

**Discussion by ibrahimbutt** [ory/kratos#3385] [Q&A]
I am interested in handling the scenario where a user has not recieved or has has lost their verification email. In this case, I expect the `createBrowserVerificationFlow` (using the JavaScript SDK with React JS) to resend an email based on [this](
https://github.com/ory/kratos/discussions/3036) discussion. On mailslurper, I only have the original email – no additonal emails are recieved.

In summary, how can I resend a verification email, in the context of a browser client?

**Comment by vinckr** [comment]
Hello @ibrahimbutt 
The verification does not get resent automatically, you can however implement this with some code on your client if you wish.

To resend a verification email, you would need to initiate a new verification flow. 
This can be done by calling the [Create Verification Flow for Browser Clients endpoint](https://www.ory.sh/docs/reference/api#tag/frontend/operation/createBrowserVerificationFlow). 

This endpoint initializes a browser-based account verification flow. Once initialized, the browser will be redirected to selfservice.flows.verification.ui_url with the flow ID set as the query parameter ?flow=.
Please note that if this endpoint is called via an AJAX request, the response contains the recovery flow without any redirects. This endpoint is not intended for API clients and only works with browsers (Chrome, Firefox, etc.).

**Reply by ibrahimbutt** [reply]
At which point is the email sent? I am already initiating a new browser verification flow, with no email being received. To add context, I am using the quickstart config modified to use my own UI.
